test/logflare/backends_test.exs

```diff
@@ -1058,4 +1058,88 @@ defmodule Logflare.BackendsTest do
       print: [configuration: false]
     )
   end
+
+  describe "sync_backend_across_cluster/1" do
+    setup do
+      insert(:plan)
+      user = insert(:user)
+      source = insert(:source, user: user)
+
+      {:ok, user: user, source: source}
+    end
+
+    test "syncs backend across cluster for all associated sources", %{
+      source: source,
+      user: user
+    } do
+      # Start actual SourceSup process
+      start_supervised!({SourceSup, source})
+      :timer.sleep(500)
```

Comment:
I noticed there's quite a few sleep calls in this test case, perhaps at some point worth reconsider testing strategy to improve performance when that becomes an issue. For example, we can add a lifecycle hook, something like

```elixir
pid = self()
start_supervised!({SourceSup, source: source, ready: fn -> send(pid, :ready) end})
assert_receive :ready
```

Another common idea is to emit Telemetry events and listen for those in tests.